buildSrc/src/main/java/gradlebuild/NativePlatformComponentPlugin.java

```diff
@@ -67,11 +71,24 @@ private void configureTestTasks(Project project) {
             Provider<String> agentName = project.getProviders().gradleProperty("agentName").forUseAtConfigurationTime();
             test.systemProperty("agentName", agentName.getOrElse("Unknown"));
         });
+
+        Stream.of("xfs", "btrfs").forEach(fileSystem -> {
+            project.getTasks().register("test" + capitalize(fileSystem), Test.class, test -> {
+                test.systemProperty(TEST_DIRECTORY_SYSTEM_PROPERTY, new File("/" + fileSystem, "native-platform/test files").getAbsolutePath());
+            });
```

Comment:
Can we check the file system type here? We can do something simple, like check the output of `findmnt -n -o FSTYPE -T /${fileSystem}`

There are more options on how to determine the file system from the path here: https://unix.stackexchange.com/questions/21799/how-can-i-determine-the-fs-type-of-my-current-working-directory

Reply:
The reason I think we should check here is that the test is far away from here. For me checking if the infrastructure is what we expect belongs to the build, not so much to a specific test that is supposed to verify if the code itself is correct.

If we want to keep the test as a unit test, I think it should be its own class, and have comments linking back to here explaining why this check is required.